src/cmd/manage_nsfs.js

```diff
@@ -385,6 +386,23 @@ async function fetch_existing_account_data(target) {
         throw err;
     }
     const data = _.merge({}, source, target);
+    if (action === ACTIONS.UPDATE) {
+        // update with user flag (=distinguished_name) on account that had uid and gid
+        const update_distinguished_name = check_if_only_distinguished_name_set(target.nsfs_account_config);
```

Comment:
why these checks are needed? shouldn't we care only about the values of the target?
Also, I suggest to avoid this code if no uid/gid/dn changed.
IIUC all you need to do is something like the suggested code
```suggestion
    const uid_update = !_.isUndefined(target.nsfs_account_config.uid);
    const gid_update = !_.isUndefined(target.nsfs_account_config.gid);
    const dn_update = !_.isUndefined(target.nsfs_account_config.distinguished_name);
    const user_fs_permissions_change = uid_update || gid_update || dn_update;
    if (action === ACTIONS.UPDATE && user_fs_permissions_change) {
        if (dn_update) {
            delete data.nsfs_account_config.uid;
            delete data.nsfs_account_config.gid;
        } else {
            delete data.nsfs_account_config.distinguished_name;
        }
    }
```

Reply:
I changed it.
I did those checks because the validation currently happens after the fetch, I wished to separate the checks between the source and the target (for example the case where someone sends `--gid` and `--user` in the update).